### ISSUE TITLE
False positive NS5000 for async Received() calls

### ISSUE THREAD

**Issue by rcdailey** [nsubstitute/NSubstitute.Analyzers#211]
Here is a test case of mine (using NUnit3):

```cs
[Test]
public async Task Repo_update_is_called_on_config_list()
{
    var repo = Resolve<IConfigTemplatesRepo>();

    // Create this to make ConfigTemplateGuideService happy. It tries to parse this file, but
    // it won't exist because we don't operate with real Git objects (so a clone never happens).
    Fs.AddFile(repo.Path.File("templates.json"), new MockFileData("{}"));

    var sut = Resolve<ConfigListCommand>();
    await sut.ExecuteAsync(default!, new ConfigListCommand.CliSettings
    {
        ListCategory = ConfigCategory.Templates
    });

    await repo.Received().Update();
}
```

The last line is the one that gives a warning. But it's perfectly valid. I think the fact that `Update()` is async is confusing it. Here's the interface in question:

```cs
public interface IConfigTemplatesRepo
{
    IDirectoryInfo Path { get; }
    Task Update();
}
```

My code is open source. The test code is [here](https://github.com/recyclarr/recyclarr/blob/9365e2649fcd8e5f7b03a8a421b77928ec3febd1/src/tests/Recyclarr.Cli.Tests/Console/Commands/ConfigCommandsIntegrationTest.cs#L40).

**Comment by tpodolak** [comment]
hi @rcdailey 
I cannot repro this issue. I've cloned your repo and there is no warning produced by analyzers
![image](https://github.com/nsubstitute/NSubstitute.Analyzers/assets/7378346/efef2f8d-5fbf-417d-adf3-c009427e3eb0)
Can not repro also in analyzers integration tests. Any chance this is IDE glitch and the warning is gone after restart/cache invalidation?

**Comment by rcdailey** [comment]
Hmm, I checked out the code at that revision and I'm not able to repro it anymore either. It's possible this was related to that analyzer issue that the .NET SDK fixed a while ago. I'm not sure. These analyzers can be a weird phenomenon sometimes...

I apologize for the false-positive and very much appreciate you taking the time to investigate.
